### PR TITLE
Improve type mapping documentation for Cassandra connector

### PR DESCRIPTION
**Pull request by tlblessing**
## Description

Improve type mapping documentation for Cassandra connector.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement to Cassandra connector documentation

> How would you describe this change to a non-technical end user or system administrator?

Added data type mapping; reformatted table; added boilerplate headings and verbiage for consistency.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 1]
```diff
@@ -292,7 +441,7 @@ configuration property to ``true``:
 
 .. _sql-delete-limitation:
 
-SQL delete limitation
+SQL DELETE limitation
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:444]: Unrelated change, please remove or make its own commit

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 2]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``UUID``
+    -
+  * - ``ROW`` with field names
+    - ``UDT``
+    -
+  * - ``VARCHAR``
+    - ``VARCHAR``
+    -
+  * - ``VARCHAR``
+    - ``VARINT``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:380]: One to many

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 3]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``TIMESTAMP``
+    -
+  * - ``UUID``
+    - ``TIMEUUID``
+    -
+  * - ``TINYINT``
+    - ``TINYINT``
+    -
+  * - ``ROW`` with anonymous fields
+    - ``TUPLE``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:368]: Put next to the other `ROW` entry to avoid confusion

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 4]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``BOOLEAN``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``DOUBLE``
+    - ``DECIMAL``
+    -
+  * - ``DOUBLE``
+    - ``DOUBLE``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:330]: One to many; one of these is wrong

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 5]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``MAP<?, ?>``
+    -
+  * - ``VARCHAR``
+    - ``SET<?>``
+    -
+  * - ``SMALLINT``
+    - ``SMALLINT``
+    -
+  * - ``VARCHAR``
+    - ``TEXT``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:356]: One to many

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 6]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+  * - ``VARCHAR``
+    - ``LIST<?>``
+    - Any collection (``LIST``, ``MAP``, or ``SET``) can be designated as
+      `FROZEN
+      <https://docs.datastax.com/en/dse/6.8/cql/cql/cql_using/refCollectionType.html>`_.
+  * - ``VARCHAR``
+    - ``MAP<?, ?>``
+    -
+  * - ``VARCHAR``
+    - ``SET<?>``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:350]: One to many

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 7]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+
+.. list-table:: Trino type to Cassandra type mapping
+  :widths: 30, 20, 50
+  :header-rows: 1
+
+  * - Trino type
+    - Cassandra database type
+    - Notes
+  * - ``VARCHAR``
+    - ``ASCII``
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:312]: One to many

**Comment by jhlodin** [comment]
Please squash commits

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 4 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:330]: Yes, need a SME to enlighten me

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 3 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:368]: Yes will redo the table since the first one is alpha order and the reverse table is not.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 8]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+  * - ``INET``
+    - ``IPADDRESS``
+    -
+  * - ``INT``
+    - ``INTEGER``
+    -
+  * - ``LIST<?>``
+    - ``VARCHAR``
+    - Any collection (``LIST``, ``MAP``, or ``SET``) can be designated as
+      `FROZEN <https://docs.datastax.com/en/dse/6.8/cql/cql/cql_using/refCollectionType.html>`_.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/cassandra.rst:257]: External links should only go to official resources. In this case, if needed, Cassandra documentation https://cassandra.apache.org/doc/latest/cassandra/cql/types.html#frozen

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 1 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst]: Removed

**Comment by tlblessing** [comment]
> Please squash commits

done

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 8 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:257]: This C* link to frozen doesn't seem to work. Will remove link.

**Review by jhlodin — APPROVED** [review]
Style looks good to me, just need @ebyhr or another SME to provide guidance on the table contents.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 9]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
-VARINT            VARCHAR
-================  ======
-
-Any collection (LIST/MAP/SET) can be designated as FROZEN, and the value is
-mapped to VARCHAR. Additionally, blobs have the limitation that they cannot be empty.
-
-Types not mentioned in the table above are not supported.
+.. _cassandra-type-mapping:
+
+Type mapping
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:203]: Could you reorder entries in both read and write mapping as https://trino.io/docs/current/language/types.html?

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 10]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Cassandra types to the corresponding Trino types according to
+the following table:
+
+.. list-table:: Cassandra type to Trino type mapping
+  :widths: 30, 20, 50
+  :header-rows: 1
+
+  * - Cassandra database type
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:221]: nit: Remove "database"

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 11]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - Notes
+  * - ``ASCII``
+    - ``VARCHAR``
+    -
+  * - ``BIGINT``
+    - ``BIGINT``
+    -
+  * - ``BLOB``
+    - ``VARBINARY``
+    - Blobs have the limitation that they cannot be empty.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:232]: Remove incorrect note.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 12]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+  * - ``INET``
+    - ``IPADDRESS``
+    -
+  * - ``INT``
+    - ``INTEGER``
+    -
+  * - ``LIST<?>``
+    - ``VARCHAR``
+    - Any collection (``LIST``, ``MAP``, or ``SET``) can be designated as
+      ``FROZEN``.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:257]: I'm not sure what you want to express in this sentence. Just removing this looks good to me.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 13]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+
+The connector maps Trino types to the corresponding Cassandra types according to
+the following table:
+
+.. list-table:: Trino type to Cassandra type mapping
+  :widths: 30, 20, 50
+  :header-rows: 1
+
+  * - Trino type
+    - Cassandra database type
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:308]: nit: Remove "database".

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 14]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``BIGINT``
+    -
+  * - ``BOOLEAN``
+    - ``BOOLEAN``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``DOUBLE``
+    - ``DECIMAL``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:321]: Remove. Trino double type is mapped to Cassandra double.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 15]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``INET``
+    -
+  * - ``REAL``
+    - ``FLOAT``
+    -
+  * - ``ROW`` with anonymous fields
+    - ``TUPLE``
+    -
+  * - ``ROW`` with field names
+    - ``UDT``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:340]: Remove as writing row type is unsupported.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 16]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    - ``SMALLINT``
+    -
+  * - ``TIMESTAMP(3) WITH TIME ZONE``
+    - ``TIMESTAMP``
+    -
+  * - ``TINYINT``
+    - ``TINYINT``
+    -
+  * - ``UUID``
+    - ``TIMEUUID``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:352]: Remove as Trino uuid type is mapped to Cassandra uuid.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 17]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    -
+  * - ``UUID``
+    - ``UUID``
+    -
+  * - ``VARBINARY``
+    - ``BLOB``
+    - Blobs have the limitation that they cannot be empty.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:359]: Remove.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 18]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    -
+  * - ``VARCHAR``
+    - ``LIST<?>``
+    - Any collection (``LIST``, ``MAP``, or ``SET``) can be designated as
+      ``FROZEN``.
+  * - ``VARCHAR``
+    - ``MAP<?, ?>``
+    -
+  * - ``VARCHAR``
+    - ``SET<?>``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:371]: Remove as Trino varchar type is mapped to Cassndra text.

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 19]
```diff
@@ -198,43 +198,192 @@ This table can then be queried in Trino::
...
+    -
+  * - ``VARCHAR``
+    - ``TEXT``
+    -
+  * - ``VARCHAR``
+    - ``VARCHAR``
+    -
+  * - ``VARCHAR``
+    - ``VARINT``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/cassandra.rst:380]: Remove.

**Review by ebyhr — COMMENTED** [review]
Could you avoid putting one-to-many entries when you're not sure about the mapping (just putting one entry is enough)? Otherwise, we have to leave many "remove" comments and it delays a review process.

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 12 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:257]: This verbiage lived underneath the table before. Are you sure we should remove it? 
image

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 11 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:232]: @ebyhr this was also original text underneath the table. Are you sure we should remove it? 

image

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 13 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:308]: done

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 15 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:340]: Done

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 16 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:352]: Done

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 14 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:321]: Done

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 17 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:359]: Done for this location, please confirm for C* -> Trino table above

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 9 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/cassandra.rst:203]: @ebyhr did my best, please review again. The original table was alpha order.

**Comment by ebyhr** [comment]
Merged, thanks!
